libpromises/expand.c

```diff
@@ -577,6 +581,13 @@ Rlist *ExpandList(EvalContext *ctx,
 
 /*********************************************************************/
 
+Rval ExpandPrivateRvalDirect(EvalContext *ctx,
```

Comment:
The introduction of this function can also go in the preparatory commit. I would just call this `ExpandPrivateRval()` and the old one, in the rare occurences that remain still, would be called like `ExpandPrivateRval(..., (Rval) {value, type})`.